requirements.txt

```diff
@@ -5,3 +5,4 @@ elasticsearch>=7.16,<7.17
 PyYAML>=6.0,<6.1
 aws_lambda_typing>=2.16.1,<3
 ujson>=5.5.0,<5.6
+pyOpenSSL>=23.0.0
```

Comment:
`pyopenssl` is in https://github.com/elastic/elastic-serverless-forwarder/blob/main/requirements-tests.txt#L13

the version is pinned to `22.0.0` because of localstack, without being able to raise the `pyopenssl` version, the alternative is to pin `cryptography==38.0.4` (https://github.com/aws/aws-sam-cli/issues/4527#issuecomment-1368871248)

I've already applied the fix on my PR preparing the 1.6.0 release (https://github.com/elastic/elastic-serverless-forwarder/pull/201/files#diff-c258030c905004296fe46761295d5c97af8641a141156e255585e968ae7a4d1dR14)

feel free to port it here so we can merge this PR and I will rebase mine